src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -121,22 +125,25 @@ export class ConnectionProfileGroup extends Disposable implements IConnectionPro
 	}
 
 	public addConnections(connections: ConnectionProfile[]): void {
-		connections.forEach((conn) => {
-			this.connections = this.connections.filter((curConn) => { return curConn.id !== conn.id; });
-			conn.parent = this;
-			this._register(conn);
-			this.connections.push(conn);
-		});
-
+		if (connections) {
+			connections.forEach((conn) => {
+				this.connections = this.connections.filter((curConn) => { return curConn.id !== conn.id; });
```

Comment:
In this case we should be setting this.connections to an empty array if it's undefined at this point and we're adding connections.

At this point though i'd almost rather we just make the public property setter not allow undefined (by setting it to an empty array if it is undefined) and then just make all these references a private property. That's a more robust way of handling it IMO.

(same with this.children below)